students/OwenGaustad/Lab4/TurnSignal/TurnSignal.c

```diff
@@ -0,0 +1,184 @@
+/***********************************************************************************************
+
+This method activates the state for my leds to shift from pin 3 to 7 first it initializes
+
+the third pin and changes its state which turns it off because I have it initiallly set to
+
+on this is unnoticable and it alowed me to have all the LED's turn on/off not the first one
+
+constantly be on I could fix this without turning the 3rd LED on but that would involve
+
+another delay and I don't want that. This method also calls the varriableDelay() twice the
+
+first delay is roughly 100ms the second takes place after they are all turned on and is 200ms
+***********************************************************************************************/
+leftTurn()  {
+  LED_PORT = 0b00001000;
+  for(led = 3; led <= 7; led++) {
+    toggleBit(LED_PORT, led);
+    variableDelay(100);
+  }
+  variableDelay(200);
+}
+/***********************************************************************************************
+
+This method activates the state for my leds to shift from pin 4 to 0 first it initializes
+
+the fourth pin and changes its state which turns it off because I have it initiallly set to
+
+on this is unnoticable and it alowed me to have all the LED's turn on/off not the first one
+
+constantly be on I could fix this without turning the 4th LED on but that would involve
+
+another delay and I don't want that. This method also calls the varriableDelay() twice the
+
+first delay is roughly 100ms the second takes place after they are all turned on and is 200ms
+***********************************************************************************************/
+rightTurn() {
+  LED_PORT = 0b00010000;
+  for(led = 4; led >= 0; led--) {
+    toggleBit(LED_PORT, led);
+    variableDelay(100);
+  }
+  variableDelay(200);
+}
+/***********************************************
+
+This method checks to see if the left button
+
+was clicked then returns 1 if yes and 0 if no
+***********************************************/
+leftBtn() {
+  if(bit_is_clear(BUTTON_PIN, BUTTON2)){
+    return 1;
+  }
+  else {
+    return 0;  
+  }
+}
+/***********************************************
+
+This method checks to see if the right button 
+
+was clicked then returns 1 if yes and 0 if no 
+***********************************************/
+rightBtn() {
+  if(bit_is_clear(BUTTON_PIN, BUTTON)){
+    return 1;
+  }
+  else {
+    return 0;  
+  }
+}
+/*******************************************************************************
+
+This method takes an integer value and stores it as a temporary variable 
+
+time Then the delay lasts for the input delay time in milliseconds              
+*******************************************************************************/
+variableDelay(time) {
+  for (delay = 0; delay <= time; delay++) {
+    _delay_ms(1);
+    if (leftBtn() == 1 | rightBtn() == 1){
+      break;
+    }
+  }
+}
+/**************
+* Main Method *
```

Comment:
BTW, it's kinda bugging me...
These are all "functions" not methods. "Method" is a term used in object oriented programming and refers to functions that are attached to objects.